WalletWasabi.Tests/UnitTests/BlockNotifierTests.cs

```diff
@@ -77,16 +79,41 @@ public async Task NotifyBlocksAsync()
 
 			// Assert that the blocks come in the right order
 			var height = 0;
-			void OnBlockInv(object s, Block b) => Assert.Equal(b.GetHash(), chain.GetBlock(height++).HashBlock);
+			string message = string.Empty;
+
+			void OnBlockInv(object blockNotifier, Block b)
+			{
+				uint256 h1 = b.GetHash();
+				uint256 h2 = chain.GetBlock(height + 1).HashBlock;
+
+				if (h1 != h2)
+				{
+					message = string.Format("height={0}, [h1] {1} != [h2] {2}", height, h1, h2);
+					cts.Cancel();
+					return;
+				}
+
+				height++;
+
+				if (height == BlockCount)
+				{
+					cts.Cancel();
+				}
+			}
+
 			notifier.OnBlock += OnBlockInv;
 
-			foreach (var n in Enumerable.Range(0, blockCount))
+			foreach (var n in Enumerable.Range(0, BlockCount))
 			{
 				await AddBlockAsync(chain);
 			}
 
 			notifier.TriggerRound();
-			await Task.Delay(TimeSpan.FromMilliseconds(100)); // give it time to process the blocks
+
+			// Wait at most 1500 ms or until cts is canceled
+			await WaitForCancelSignalAsync(TimeSpan.FromMilliseconds(1500), cts.Token);
```

Comment:
This is not very important to me in the tests code but wherever we have an `await` we use `ConfigureAwait(false)` and it is missing in all this test.
I think it is easier to thing in terms of seconds, that's why I would prefer to see `FromSeconds(1.5)` than `FromMilliseconds(1500)`

Reply:
Fixed.